pkg/registry/core/pod/storage/eviction.go

```diff
@@ -138,7 +138,12 @@ func (r *EvictionREST) Create(ctx context.Context, obj runtime.Object, createVal
 	// At this point there was either no PDB or we succeeded in decrementing
 
 	// Try the delete
-	_, _, err = r.store.Delete(ctx, eviction.Name, eviction.DeleteOptions)
+	deletionOptions := eviction.DeleteOptions
+	if deletionOptions == nil {
+		// default to non-nil to trigger graceful deletion
+		deletionOptions = &metav1.DeleteOptions{}
```

Comment:
if deletionOpt is nil, we're going to use default grace period?

Reply:
eviction without delete options would behave like delete without delete options

This is a behavior change. We need to decide if eviction (which is intended to allow safe deletion of pods according to pod disruption budget) essentially force deleting pods by default is a problem worth fixing. 

Notably, kubectl drain never passes nil delete options, which is good. https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/drain/drain.go#L526)

Reply:
> This is a behavior change.

Yes; but it seems a "correction" instead of "change" considering "delete without options" :)

Reply:
> Yes; but it seems a "correction" instead of "change" considering "delete without options" :)

I don't disagree, but would like an ack from @kubernetes/api-approvers

Reply:
I agree.